tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenTestFactory.cs

```diff
@@ -21,6 +21,21 @@
 
 namespace MongoDB.Driver.Tests.JsonDrivenTests
 {
+    public class JsonDrivenTestsContext
+    {
```

Comment:
Ideally, I would add the other properties like collectionName and similar here, but leaving it as is for now to reduce the number of git changes

Reply:
NOTE: this was added because now we need to save the tests state between tests steps

Reply:
I don't think this class is right.

For one, there is some confusion about how the word context is used. Is this a context, or a context of contexts or  a dictionary of contexts?

Also, why can't we make this dictionary a property of IJsonDrivenTestRunner, or perhaps use the existing objectMap for this?

Reply:
We can discuss it one more time